### PR TITLE
Add training call argument for MultiHeadAttention

### PR DESCRIPTION
**Pull request by WindQAQ**
Miss `training` call argument for dropout layer. /cc @tanzhenyu for visibility.

### CODE REVIEW

**Comment by WindQAQ** [comment]
Hi @tanzhenyu, sorry for bothering you. Can you review this when time allows? We are going to deprecate `MultiHeadAttention` in addons, and this will give us a full spectrum and more stable API during deprecation.

**Review by tanzhenyu — APPROVED** [review]
Thank you!

**Review by gbaned — APPROVED** [review]

**Comment by saberkun** [comment]
Is the training argument required? I thought keras will automatically inject training from models.
Something would not work if it is inside a tf.module?

**Comment by WindQAQ** [comment]
> Is the training argument required? I thought keras will automatically inject training from models.
> Something would not work if it is inside a tf.module?

Hi @saberkun It will be injected when using it inside `tf.keras.Model` or `tf.keras.layers.Layer` but will not work inside `tf.Module`. I think this is not an usually case actually.
